examples/miscellaneous/plot_outlier_detection_bench.py

```diff
@@ -17,181 +19,361 @@
 """
 
 # Author: Pharuj Rajborirug <pharuj.ra@kmitl.ac.th>
+#         Arturo Amor <david-arturo.amor-quiroz@inria.fr>
 # License: BSD 3 clause
 
-print(__doc__)
+# %%
+# Datasets preprocessing and model training
+# =========================================
+#
+# This example uses real-world datasets available in :class:`sklearn.datasets`.
+# Due to computational constraints of the scikit-learn documentation, the sample
+# size of some datasets is reduced using a stratified
+# :class:`~sklearn.model_selection.train_test_split`.
+# After the data preprocessing, the datasets' targets will have two classes, 0
+# representing inliers and 1 representing outliers.
+#
+# Different datasets require different preprocessing. One has to take into
+# account that tree based models such as
+# :class:`~sklearn.ensemble.IsolationForest` can deal with categorical variables
+# encoded using an :class:`~sklearn.preprocessing.OrdinalEncoder`, whereas
+# neighbors based models such as :class:`~sklearn.neighbors.LocalOutlierFactor`
+# require a :class:`~sklearn.preprocessing.OneHotEncoder`.
+#
+# Neighbors based models may also require scaling of the numerical features (see
+# for instance :ref:`neighbors_scaling`). In the presence of outliers, a good
+# option is to use a :class:`~sklearn.preprocessing.RobustScaler`.
+#
+# The following `fit_predict` function returns average outlier score of X.
+
+from sklearn.neighbors import LocalOutlierFactor
+from sklearn.ensemble import IsolationForest
+from sklearn.preprocessing import (
+    OneHotEncoder,
+    OrdinalEncoder,
+    RobustScaler,
+)
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import make_pipeline
+from time import perf_counter
+
+
+def fit_predict(X, model_name, categorical_columns=(), n_neighbors=20):
+
```

Comment:
Is this empty line on purpose?

Reply:
Erased the empty line, thanks for the comment ;)